bitnami/postgresql-repmgr/16/debian-12/rootfs/opt/bitnami/scripts/librepmgr.sh

```diff
@@ -625,15 +625,25 @@ repmgr_clone_primary() {
         rm -rf "$waldir" && ensure_dir_exists "$waldir"
     fi
 
+    if am_i_root; then
+        chown -R "$POSTGRESQL_DAEMON_USER:$POSTGRESQL_DAEMON_GROUP" "$POSTGRESQL_DATA_DIR"
```

Comment:
Ahh yes, it is necessary for the standby nodes since the function that executes the line you mention above, `postgresql_initialize()`, is called **after**  the `repmgr_clone_primary()` function for the standby nodes...

reference below, the `repmgr_rewind()` function ultimately calls `repmgr_clone_primary()`:
https://github.com/bitnami/containers/blob/c4fcee416cc319efbde9b6e603f510e06ed602fc/bitnami/postgresql-repmgr/16/debian-12/rootfs/opt/bitnami/scripts/librepmgr.sh#L828-L832

Perhaps it makes more sense to move this entire block into the `repmgr_initialize()` function instead to ensure it always runs at the beginning of the setup?

https://github.com/bitnami/containers/blob/c4fcee416cc319efbde9b6e603f510e06ed602fc/bitnami/postgresql-repmgr/16/debian-12/rootfs/opt/bitnami/scripts/libpostgresql.sh#L615-L619

Reply:
Ok, thanks for clarifying it, I see!

> Perhaps it makes more sense to move this entire block into the `repmgr_initialize()` function instead to ensure it always runs at the beginning of the setup?

We shouldn't alter `libpostgresql.sh` if possible because it is being used in multiple containers.

Maybe we could just move the chown for `"$POSTGRESQL_DATA_DIR"` to `repmgr_initialize` as you mention, next to this other ownership change: https://github.com/bitnami/containers/blob/b94ed19206a604729cad250e307c60a4a7068c9e/bitnami/postgresql-repmgr/16/debian-12/rootfs/opt/bitnami/scripts/librepmgr.sh#L826